azerothcore/azerothcore-wotlk
# Add (Core\Player): Added createplayerinfo_cast_spell support cast spell for some class spells

(Warrior's Battle Stance and Death Knight's Blood Presence) on first login.



## Changes Proposed:
-  Implement TC's playercreateinfo_cast_spell system and tables
-  Added support cast spell for some class spells (Warrior's Battle Stance and Death Knight's Blood Presence) on first login.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/9439

## SOURCE:

https://github.com/TrinityCore/TrinityCore/commit/c8eb69df10b36d19a77743a037e700b001c8ed77
https://github.com/TrinityCore/TrinityCore/commit/30313fba8dbf5dbcdf4e91e1ea179ad1b8c9bdc5

## Tests Performed:

- Builds, Compiles, and Runs fine.

## How to Test the Changes:


1. Login with new created character class Death Knight or Warrior
2. Ensure DK has Blood Presence
3. Ensure Warrior has Battle Stance

## Known Issues and TODO List:


- [x] Independent Testing


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
